src/SmartComponents/RunTaskModal/SystemsSelect.js

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from 'react';
+import propTypes from 'prop-types';
+import { Alert } from '@patternfly/react-core/dist/js/components/Alert';
+import { Flex, FlexItem } from '@patternfly/react-core/dist/js/layouts/Flex';
+import {
+  Form,
+  FormGroup,
+} from '@patternfly/react-core/dist/js/components/Form';
+import { TextInput } from '@patternfly/react-core/dist/js/components/TextInput';
+import { ValidatedOptions } from '@patternfly/react-core/dist/js/helpers/constants';
+import SystemTable from '../SystemTable/SystemTable';
+import {
+  AVAILABLE_TASKS_ROOT,
+  INFO_ALERT_SYSTEMS,
+  TASKS_API_ROOT,
+} from '../../constants';
+import ReactMarkdown from 'react-markdown';
+import { fetchSystems } from '../../../api';
+import warningConstants from '../warningConstants';
+
+const SystemsSelect = ({
+  createTaskError,
+  description,
+  selectedIds,
+  setSelectedIds,
+  setTaskName,
+  slug,
+  taskName,
+}) => {
+  const warningConstantMapper = `${slug
+    ?.toUpperCase()
+    .replace(/-/g, '_')}_WARNING`;
+
+  const [filterSortString, setFilterSortString] = useState('');
+
+  const bulkSelectIds = async (type, options) => {
+    let newSelectedIds = [...selectedIds];
+
+    switch (type) {
+      case 'none': {
+        setSelectedIds([]);
+        break;
+      }
+
+      case 'page': {
+        options.items.forEach((item) => {
+          if (!newSelectedIds.includes(item.id)) {
+            newSelectedIds.push(item.id);
+          }
+        });
+
+        setSelectedIds(newSelectedIds);
+        break;
+      }
+
+      case 'all': {
+        let results = await fetchSystems(filterSortString);
+        setSelectedIds(results.data.map(({ id }) => id));
+        break;
+      }
+    }
+  };
+
+  const selectIds = (_event, _isSelected, _index, entity) => {
+    let newSelectedIds = [...selectedIds];
+
+    !newSelectedIds.includes(entity.id)
+      ? newSelectedIds.push(entity.id)
+      : newSelectedIds.splice(newSelectedIds.indexOf(entity.id), 1);
+
+    setSelectedIds(newSelectedIds);
+  };
+
```

Comment:
This can probably be extracted into a hook that returns the two handler functions.